structured/privilege.proto

```diff
@@ -0,0 +1,58 @@
+// Copyright 2015 The Cockroach Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+// implied. See the License for the specific language governing
+// permissions and limitations under the License. See the AUTHORS file
+// for names of contributors.
+//
+// Author: Marc Berhault (marc@cockroachlabs.com)
+
+syntax = "proto2";
+package cockroach.structured;
+option go_package = "structured";
+
+import "gogoproto/gogo.proto";
+
+option (gogoproto.sizer_all) = true;
+option (gogoproto.marshaler_all) = true;
+option (gogoproto.unmarshaler_all) = true;
+
+// This list must match the list of privileges in sql/parser/sql.y.
+enum Privilege {
+  // ALL is specifically encoded so that it will automatically
+  // pick up new privileges.
+  ALL    = 0;
```

Comment:
Wondering if `ALL` should be non-zero. Seems error prone for the zero value to get all privileges.